workflow/workflow_test.go

```diff
@@ -308,7 +308,6 @@ func TestWithDoubleSerializationDetection(t *testing.T) {
 		},
 	}
 	for i, testData := range testIter {
```

Comment:
Ah; lint dislikes that you're not using `i`.

Reply:
Oops. Fixed.

Reply:
Well, you got past the linter this time, but the tests remain unhappy about something...

Reply:
It appears that the test make an existing issue more likely to trigger. I'm looking into this, and experimenting with waiting with a timeout instead of no waiting at all.

Reply:
There was a race condition in how the waitgroup was used. I pushed a change that fixes it. It's a one-line change.